src/server/authserver/Server/AuthSession.cpp

```diff
@@ -415,8 +419,8 @@ void AuthSession::LogonChallengeCallback(PreparedQueryResult result)
     {
         Trinity::Crypto::SRP6::Salt salt;
         Trinity::Crypto::SRP6::Verifier verifier;
-        HexStrToByteArray(fields[11].GetString(), salt, true); /* this is actually flipped in the DB right now, old core did hexstr (big endian) -> bignum -> byte array (little-endian) */
-        HexStrToByteArray(fields[12].GetString(), verifier);
+        fields[11].GetBinary(salt);
```

Comment:
Can you use the other overload here (and everywhere else), this looks just weird/wrong